pkg/dbmate/migrations_test.go

```diff
@@ -0,0 +1,50 @@
+package dbmate
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseMigrationContents(t *testing.T) {
+	migration := `-- migrate:up
+create table users (id serial, name text);
+-- migrate:down
+drop table users;`
+
+	up, down := parseMigrationContents(migration)
+
+	require.Equal(t, "-- migrate:up\ncreate table users (id serial, name text);", up.Contents)
+	require.Equal(t, false, up.Options.SkipTransaction())
+
+	require.Equal(t, "-- migrate:down\ndrop table users;", down.Contents)
+	require.Equal(t, false, down.Options.SkipTransaction())
+
+	migration = `-- migrate:down
+drop table users;
+-- migrate:up
+create table users (id serial, name text);
+`
+
+	up, down = parseMigrationContents(migration)
+
+	require.Equal(t, "-- migrate:up\ncreate table users (id serial, name text);", up.Contents)
+	require.Equal(t, false, up.Options.SkipTransaction())
+
+	require.Equal(t, "-- migrate:down\ndrop table users;", down.Contents)
+	require.Equal(t, false, down.Options.SkipTransaction())
+
+	migration = `-- migrate:up skip_transaction:true
```

Comment:
maybe add a comment here saying that the reason you chose this particular operation is that it will fail inside a transaction

Reply:
can do